src/main/java/org/traccar/protocol/EgtsProtocolDecoder.java

```diff
@@ -172,12 +189,14 @@ protected Object decode(
                         buf.readUnsignedShortLE(); // home dispatcher identifier
                     }
                     if (BitUtil.check(flags, 1)) {
-                        getDeviceSession(
-                                channel, remoteAddress, buf.readSlice(15).toString(StandardCharsets.US_ASCII).trim());
+                        String imei = buf.readSlice(15).toString(StandardCharsets.US_ASCII).trim();
+                        LOGGER.trace("[{}] Using IMEI1 as deviceId: {}", channel == null ? "" : channel.id(), imei);
```

Comment:
I don't really understand the need for logging here. I would recommend reverting all 3 places.